### PR TITLE
release: use GitHub actions

### PR DESCRIPTION
**Pull request by v1v**
### What

Use GitHub actions in conjunction with the Buildkite GPG singing pipeline for the Release

### Why

Remove support for Jenkins

### Implementation details

GPG signing runs in Buildkite, that's the reason to use a Google Bucket to pass the unsinged binaries so they can be signed afterwards.

### UI

image

### Test

I ran this [build](https://github.com/elastic/apm-agent-php/actions/runs/6147424970) using the test branch -> https://github.com/elastic/apm-agent-php/tree/test/migrate-jenkins

```diff
diff --git a/.github/workflows/release.yml b/.github/workflows/release.yml
index 568b0bfb..ffbe2d5f 100644
--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -5,6 +5,8 @@ permissions:
   contents: read
 
 on:
+  ### TODO: For testing purposes
+  workflow_dispatch: ~
   create:
     tags: [ "v*" ]
 
@@ -110,14 +112,14 @@ jobs:
 
       - name: Create draft release
         run: |-
-          make -f .ci/Makefile draft-release
+          echo "make -f .ci/Makefile draft-release"
           echo "RELEASE_ID=$(make -f .ci/Makefile get-draft-release)" >> $GITHUB_ENV
 
       - name: Verify draft release
-        run: ORIGINAL_PACKAGES_LOCATION=build/packagesmake -f .ci/Makefile download-verify
+        run: echo "ORIGINAL_PACKAGES_LOCATION=build/packagesmake -f .ci/Makefile download-verify"
 
       - name: Publish release
-        run: make -f .ci/Makefile github-release-ready
+        run: echo "make -f .ci/Makefile github-release-ready"
 
   notify:
     if: always()
@@ -140,5 +142,5 @@ jobs:
           vaultUrl: ${{ secrets.VAULT_ADDR }}
           vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
           vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
-          slackChannel: "#apm-agent-php"
+          slackChannel: "#observablt-bots"
           message: "[${{ github.repository }}] Release (<${{ github.server_url }}/${{ github.repository }}/releases/tag/${{ github.ref_name }}|${{ github.ref_name }}>)"

```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,10 +5,10 @@ permissions:
   contents: read
 
 on:
+  ### TODO: For testing purposes
   workflow_dispatch:
```
Comment (v1v) [inline .github/workflows/release.yml:9]:
```suggestion
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -58,62 +59,230 @@ jobs:
...
+      - name: Unzip signed packages
+        run: |-
+          cd build/packages
+          unzip signed-artifacts.zip
+
+      ## TODO: uncomment
+      - name: Create draft release
+        run: |-
+          echo "make -f .ci/Makefile draft-release"
+          echo "RELEASE_ID=$(make -f .ci/Makefile get-draft-release)" >> $GITHUB_ENV
```
Comment (v1v) [inline .github/workflows/release.yml:254]:
```suggestion
      - name: Create draft release
        run: |-
          make -f .ci/Makefile draft-release
          echo "RELEASE_ID=$(make -f .ci/Makefile get-draft-release)" >> $GITHUB_ENV
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -58,62 +59,230 @@ jobs:
...
 
-      - if: ${{ failure() }}
-        uses: elastic/apm-pipeline-library/.github/actions/slack-message@current
+      ## TODO: uncomment
+      - name: Verify draft release
+        run: echo "ORIGINAL_PACKAGES_LOCATION=build/packagesmake -f .ci/Makefile download-verify"
+
+      ## TODO: uncomment
+      - name: Publish release
+        run: echo "make -f .ci/Makefile github-release-ready"
```
Comment (v1v) [inline .github/workflows/release.yml:262]:
```suggestion
      - name: Verify draft release
        run: ORIGINAL_PACKAGES_LOCATION=build/packagesmake -f .ci/Makefile download-verify

      - name: Publish release
        run: make -f .ci/Makefile github-release-ready
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -58,62 +59,230 @@ jobs:
...
+          needs: ${{ toJSON(needs) }}
+      - uses: elastic/apm-pipeline-library/.github/actions/notify-build-status@current
+        with:
+          status: ${{ steps.check.outputs.status }}
+          vaultUrl: ${{ secrets.VAULT_ADDR }}
+          vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
+          vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
+          ## TODO: uncomment
+          #slackChannel: "#apm-agent-php"
+          slackChannel: "#on-week-oblt-productivity"
```
Comment (v1v) [inline .github/workflows/release.yml:287]:
```suggestion
          slackChannel: "#apm-agent-php"
```

**Review by intuibase — DISMISSED** [review]

**Review by reakaleek — DISMISSED** [review]

**Comment by v1v** [comment]
I'm gonna refactor this PR so it can use the changes in https://github.com/elastic/apm-agent-php/pull/1058

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -5,115 +5,140 @@ permissions:
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    env:
+      GITHUB_TOKEN: ${{ github.token }}
+      TAG_NAME: ${{ github.ref_name }}
+    steps:
 
-      - run: echo "Publish draft release"
+      - uses: actions/download-artifact@v3
```
Comment (v1v) [inline .github/workflows/release.yml:101]:
```suggestion
      - uses: actions/checkout@v3
      - uses: actions/download-artifact@v3
```

**Review by amannocci — APPROVED** [review]

**Pull request merged**